### PR TITLE
Fixes # 3592 - Clarify Fixtures' Documentation

### LINKED ISSUE #3592: Fixture docs - possible point of confusion
Very minor gripe here, but I lost some time over it and I wanted to give feedback :)

https://docs.pytest.org/en/latest/fixture.html

I got very confused in this page. 
TLDR: Variable naming in examples made it difficult for me to figure out how to adapt examples to my own use case

```

# content of conftest.p
import pytest
import smtplib

@pytest.fixture(scope="module")
def smtp():
    return smtplib.SMTP("smtp.gmail.com", 587, timeout=5)

# content of test_module.py

def test_ehlo(smtp):
    response, msg = smtp.ehlo()
    assert response == 250
    assert b"smtp.gmail.com" in msg
    assert 0  # for demo purposes

def test_noop(smtp):
    response, msg = smtp.noop()
    assert response == 250
    assert 0  # for demo purposes
```

If you don't know anything about the smtp module, example reads as if you have to call a method on "smtp" that matches your test name (minus test_).

When looking at an example, a key thing that helps is if you can quickly discover what is arbitrary, and what "has to match something else".

When writing examples, I try to make arbitrary names look arbitrary by munging them in some way or making them look frivolous.

Eg, is smtp here an arbitrary name, or is it something special? smtp is a general protocol, so it's strange to use it as a variable name. It's also almost a module name. Protocols don't tend to "do things",

I would do something like this.
```

# content of conftest.p
import pytest
import smtplib

@pytest.fixture(scope="module")
def my_fixture():
    return smtplib.SMTP("smtp.gmail.com", 587, timeout=5)

# content of test_module.py

def test_something(my_fixture):
    response, msg = my_fixture.ehlo()
    assert response == 250
    assert b"smtp.gmail.com" in msg
    assert 0  # for demo purposes

def test_anotherthing(my_fixture):
    response, msg = my_fixture.noop()
    assert response == 250
    assert 0  # for demo purposes
```

### PR DESCRIPTION
**Pull request by caramelomartins**
Hey,

I'm starting to use `pytest` and thought I'd give a small help. I've noticed #3592 hasn't been fixed yet. I decided to go the suggested way with `smtp_connection` because it felt clearer. This PR only features documentation changes.

Since the page was quite large and I am inexperienced, I'll review it again tomorrow but this gives an opportunity for sharing what has already been done.

Fixes #3592

Checklist:

- [x] Create a new changelog file in the `changelog` folder, with a name like `<ISSUE NUMBER>.<TYPE>.rst`. See [changelog/README.rst](/changelog/README.rst) for details.
- [x] Target the `master` branch for bug fixes, documentation updates and trivial changes.

Not Applicable:

- [ ] Target the `features` branch for new features and removals/deprecations.
- [ ] Include documentation when adding new features.
- [ ] Include new tests or update existing tests when applicable.

I didn't add myself to `AUTHORS` because I don't believe it warrants that. 

Hope everything is okay. Let me know if changes are needed.

Regards,
Hugo

### CODE REVIEW

**Comment by caramelomartins** [comment]
I see the builds have failed. Will take a look at why `linting` failed.

**Comment by nicoddemus** [comment]
Hi @caramelomartins, thanks for the contribution.

You can run the linting checks by installing pre-commit, see the [contributing guide](https://github.com/pytest-dev/pytest/blob/master/CONTRIBUTING.rst#short-version) for details. 👍

**Comment by nicoddemus** [comment]
Thanks @caramelomartins, but I think there are a few spots where it still uses `smtp` instead of `smtp_connection`. If you are on linux, could you please run `tox -e regen` to generate the examples again? The problems will be clearer that way.

**Comment by caramelomartins** [comment]
@nicoddemus I'll take a look. There were some that I left because I thought they made sense but I'll review it again (using `tox -e regn` like you suggest).

**Comment by nicoddemus** [comment]
Thanks @caramelomartins! I took the liberty of running pre-commit in order to fix linting on CI. I will review the rest of the PR as soon as I can. 👍

**Comment by caramelomartins** [comment]
@nicoddemus Thank you for running the linter. :medal_sports:  Let me know if you find any more troubles. :+1:

File: doc/en/fixture.rst [hunk 1]
```diff
@@ -456,8 +457,8 @@ server URL in its module namespace::
 
     smtpserver = "mail.python.org"  # will be read by smtp fixture
 
-    def test_showhelo(smtp):
-        assert 0, smtp.helo()
+    def test_showhelo(smtp_connection):
```
Comment (nicoddemus) [inline doc/en/fixture.rst]: This code seems right, but is raising a `NameError` below, not sure why...

File: doc/en/fixture.rst [hunk 2]
```diff
@@ -568,57 +568,48 @@ So let's just do another run::
...
+    smtp_connection = <function smtp_connection at 0xdeadbeef>
 
-        def test_ehlo(smtp):
-            response, msg = smtp.ehlo()
-            assert response == 250
-            assert b"smtp.gmail.com" in msg
-    >       assert 0  # for demo purposes
-    E       assert 0
+        def test_ehlo(smtp_connection):
+    >       response, msg = smtp_connection.ehlo()
```
Comment (nicoddemus) [inline doc/en/fixture.rst]: Here it is also weird, the fixture value is actually the function...

File: doc/en/fixture.rst [hunk 3]
```diff
@@ -568,57 +568,48 @@ So let's just do another run::
...
-    smtp = <smtplib.SMTP object at 0xdeadbeef>
+    smtp_connection = <function smtp_connection at 0xdeadbeef>
 
-        def test_noop(smtp):
-            response, msg = smtp.noop()
-            assert response == 250
-    >       assert 0  # for demo purposes
-    E       assert 0
+        def test_noop(smtp_connection):
+    >       response, msg = smtp_connection.noop()
```
Comment (nicoddemus) [inline doc/en/fixture.rst]: Here it is also weird, the fixture value is actually the function...

File: doc/en/fixture.rst [hunk 4]
```diff
@@ -568,57 +568,48 @@ So let's just do another run::
...
-    smtp = <smtplib.SMTP object at 0xdeadbeef>
+    smtp_connection = <function smtp_connection at 0xdeadbeef>
 
-        def test_ehlo(smtp):
-            response, msg = smtp.ehlo()
-            assert response == 250
-    >       assert b"smtp.gmail.com" in msg
-    E       AssertionError: assert b'smtp.gmail.com' in b'mail.python.org\nPIPELINING\nSIZE 51200000\nETRN\nSTARTTLS\nAUTH DIGEST-MD5 NTLM CRAM-MD5\nENHANCEDSTATUSCODES\n8BITMIME\nDSN\nSMTPUTF8'
+        def test_ehlo(smtp_connection):
+    >       response, msg = smtp_connection.ehlo()
```
Comment (nicoddemus) [inline doc/en/fixture.rst]: Here it is also weird, the fixture value is actually the function...

File: doc/en/fixture.rst [hunk 5]
```diff
@@ -568,57 +568,48 @@ So let's just do another run::
...
-    smtp = <smtplib.SMTP object at 0xdeadbeef>
+    smtp_connection = <function smtp_connection at 0xdeadbeef>
 
-        def test_noop(smtp):
-            response, msg = smtp.noop()
-            assert response == 250
-    >       assert 0  # for demo purposes
-    E       assert 0
+        def test_noop(smtp_connection):
+    >       response, msg = smtp_connection.noop()
```
Comment (nicoddemus) [inline doc/en/fixture.rst]: Here it is also weird, the fixture value is actually the function...

**Comment by caramelomartins** [comment]
@nicoddemus I'll explore the problems you highlight once I can. :+1:

**Comment by nicoddemus** [comment]
@caramelomartins thanks! I've been meaning to take a look myself, but didn't find the time yet. 👍

**Comment by nicoddemus** [comment]
Hi @caramelomartins,

Found the problem, it was just a print that was using the old name (4dfe2eee945db025aad942fef71b98b4f789c349).

I regenerated the docs myself and now this is good to go. 

Thanks again for the PR!

**Comment by caramelomartins** [comment]
Ah, shame I couldn't catch it @nicoddemus .

Thanks for the patience and for the awesome work on pytest!

**Pull request merged**